ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/commands/ApplyMigrationCommand.java

```diff
@@ -248,6 +253,10 @@ private boolean apply(
           if (migration.getVersion() < minimumVersion) {
             return false;
           }
+          if (MetadataUtil
+              .isVersionMigrated(String.valueOf(migration.getVersion()), config, ksqlClient)) {
+            return false;
```

Comment:
Also filters here, in case someone has migrations 1, 2 and 3 and calls `apply -v 3`, `apply -v 1` and then `apply -a`

Reply:
I thought we decided not to allow users to apply migrations out of order? As in, once the user has run `apply -v 3` then they should not be allowed to run `apply -v 1`. Do we not currently have a check for this? If not it was probably missed as part of the same gap that missed applying a migration version that has already been applied, and we should conversation about whether we want to start enforcing this or not. (My vote is yes, but a separate PR is fine rather than including that change in this one.)

Reply:
We don't have that check yet. If we want to enforce in-order only, then this check actually becomes obsolete, since we would strictly require users to apply versions greater than what was last applied. I'll just close this PR in favour of that.

Reply:
Yes, I think we should enforce that users only apply migrations in increasing order (not necessarily consecutive). See previous discussion here: https://github.com/confluentinc/ksql/pull/7099#discussion_r584892827
If we add this check into the apply command but not anything to do with validation, then we should be fine from a backwards compatibility standpoint.